File: src/jvmMain/kotlin/com/dropbear/components/CustomPropertiesNative.java
package com.dropbear.components;

import com.dropbear.EucalyptusCoreLoader;
import com.dropbear.math.Vector3d;

public class CustomPropertiesNative {
    static {
        new EucalyptusCoreLoader().ensureLoaded();
    }

    public static native boolean customPropertiesExistsForEntity(long worldHandle, long entityId);

    public static native String getStringProperty(long worldHandle, long entityId, String label);
    public static native Integer getIntProperty(long worldHandle, long entityId, String label);
    public static native Long getLongProperty(long worldHandle, long entityId, String label);
    public static native Double getDoubleProperty(long worldHandle, long entityId, String label);
    public static native Float getFloatProperty(long worldHandle, long entityId, String label);
    public static native Boolean getBoolProperty(long worldHandle, long entityId, String label);
    public static native Vector3d getVec3Property(long worldHandle, long entityId, String label);

    public static native void setStringProperty(long worldHandle, long entityId, String label, String value);
    public static native void setIntProperty(long worldHandle, long entityId, String label, int value);
    public static native void setLongProperty(long worldHandle, long entityId, String label, long value);
    public static native void setFloatProperty(long worldHandle, long entityId, String label, double value);
    public static native void setBoolProperty(long worldHandle, long entityId, String label, boolean value);
    public static native void setVec3Property(long worldHandle, long entityId, String label, Vector3d value);
}

File: src/nativeMain/kotlin/com/dropbear/components/CustomProperties.native.kt
package com.dropbear.components

import com.dropbear.EntityId
import com.dropbear.math.Vector3d

internal actual fun CustomProperties.getStringProperty(
    entityHandle: Long,
    label: String
): String? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getIntProperty(entityHandle: Long, label: String): Int? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getLongProperty(
    entityHandle: Long,
    label: String
): Long? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getDoubleProperty(
    entityHandle: Long,
    label: String
): Double? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getFloatProperty(
    entityHandle: Long,
    label: String
): Float? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getBoolProperty(
    entityHandle: Long,
    label: String
): Boolean? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.getVec3Property(
    entityHandle: Long,
    label: String
): Vector3d? {
    TODO("Not yet implemented")
}

internal actual fun CustomProperties.setStringProperty(
    entityHandle: Long,
    label: String,
    value: String
) {
}

internal actual fun CustomProperties.setIntProperty(
    entityHandle: Long,
    label: String,
    value: Int
) {
}

internal actual fun CustomProperties.setLongProperty(
    entityHandle: Long,
    label: String,
    value: Long
) {
}

internal actual fun CustomProperties.setFloatProperty(
    entityHandle: Long,
    label: String,
    value: Double
) {
}

internal actual fun CustomProperties.setBoolProperty(
    entityHandle: Long,
    label: String,
    value: Boolean
) {
}

internal actual fun CustomProperties.setVec3Property(
    entityHandle: Long,
    label: String,
    value: Vector3d
) {
}

internal actual fun customPropertiesExistsForEntity(entityId: EntityId): Boolean {
    TODO("Not yet implemented")
}